processing/src/test/java/org/apache/druid/segment/IndexBuilder.java

```diff
@@ -222,15 +223,31 @@ public QueryableIndex buildMMappedIndex()
     Preconditions.checkNotNull(indexMerger, "indexMerger");
     Preconditions.checkNotNull(tmpDir, "tmpDir");
     try (final IncrementalIndex incrementalIndex = buildIncrementalIndex()) {
+      List<IndexableAdapter> adapters = Collections.singletonList(
+          new QueryableIndexIndexableAdapter(
+              indexIO.loadIndex(
+                  indexMerger.persist(
+                      incrementalIndex,
+                      new File(
+                          tmpDir,
+                          StringUtils.format("testIndex-%s", ThreadLocalRandom.current().nextInt(Integer.MAX_VALUE))
+                      ),
+                      indexSpec,
+                      null
+                  )
+              )
+          )
+      );
+      // still merge it since that follows the normal path of persist then merge
```

Comment:
nit: "still merge" implies that this comment is referring to a change.  A new reader is not going to know what that change is...  It looks like you are exercising the behavior of what happens when it reads back over the segment to persist a new one?  Perhaps changing this to 

> Do a merge, which will do yet another persist and load again to validate that the behavior of writing and then
> reading still does good things

This is gonna have performance implications for test run times too, I fear.  But, if we only ever do this once for each data set that we are indexing, it shouldn't be good expensive...

Reply:
> This is gonna have performance implications for test run times too, I fear. But, if we only ever do this once for each data set that we are indexing, it shouldn't be good expensive...

yeah, I was concerned about that, but it looks like it hasn't made the (already terrible) processing tests times any worse, so I think its worth it because of the extra coverage of ensuring both indexable adapters are flexed when building test data segments (and more closely matches current ingest task behavior)